### PR TITLE
documented error responses in drf_yasg schema and removed unused imports

### LINKED ISSUE #405: Document error schemas
In API v1, we override a lot of the default behavior for exceptions in Django REST Framework. The possible error responses should be clearly defined in our OpenAPI so consumers can anticipate the situations that they will arise. This should cover validation error and error 500s in particular.

See the `@swagger_auto_schema` decorator in `image_views.py` for an example of how this is done. The simplest way is to define Django Rest Framework serializers that match the structure of the error messages.

### PR DESCRIPTION
**Pull request by Vineet-Sharma29**
Fixes #405 #436

In order to document errors in `drf_yasg` following changes are made:-
- added documentation for `500` error in all views of `api` app
- in `image_views.py` added serialiser to `ImageDetail`'s `get` and `delete` methods so as to document `400` error
- in `list_views.py` added serialiser to `CreateList`'s `post` method and `ListDetail`'s `get`, `delete` and `put` methods to document `400` error
- in `site_views.py` added serialiser to `Register`'s `post` method to document `400` error
- in `link_views.py` added serialiser to `CreateShortenedLink`'s `post` method to document `400` error

Following new serializers have been added:-
- in `image_serializers.py`, `ImageDetailInputErrorSerializer` has been added to serialize `uuid` field
- in `list_serializers.py`, `ListDetailInputErrorSerializer` to serialize `slug` field

Removed unused import from:-
- `analytics/tests.py`
- `cccatalog-api/cccatalog/api/controllers/search_controller.py`
- `cccatalog-api/cccatalog/api/serializers/image_serializers.py`
- `cccatalog-api/cccatalog/api/views/image_views.py`
- `cccatalog-api/cccatalog/api/views/link_views.py`
- `cccatalog-api/cccatalog/scripts/thumbnail_load_test/locustfile.py`
- `cccatalog-api/cccatalog/urls.py`
- `ingestion_server/ingestion_server/qa.py`
- `ingestion_server/test/unit_tests.py`
- `misc/gen_popularity.py`
---

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

**Comment by Vineet-Sharma29** [comment]
@aldenstpage I am not sure regarding following:-
- `/v1/auth_tokens/register` takes `name`, `description` and `email` as args and `OAuth2RegistrationSerializer` serialize these fields. So, where should it be changed?
- `v1/images` takes query parameter as serialized by `OAuth2RegistrationSerializer`. Therefore `400` can come. Why should it not be considered?
- what format for error `500` should be used?

**Comment by aldenstpage** [comment]
Some serializers model requests, and others model responses, and you've switched them around in some cases. For instance, for `/v1/auth_tokens/register`, you say that error 400 returns a response like this:
```
{
    "name": "string",
    "description": "string",
    "email": "user@example.com"
}
```

But it doesn't. There are similar issues with the other endpoints I pointed out.

**Comment by Vineet-Sharma29** [comment]
@aldenstpage apologies for late reply. I have fixed following in codebase:-
- changed error `500` formats in `SearchImages` `get` method, `ImageDetail` `get` method, `CreateShortenedLink` `post` method, `ImageStats` `get` method and `register` `post` method
- removed error `400` from `ImageDetail`'s `get` method and `500` from `ImageDetail`'s `delete` method
- added error `401` in `ImageDetail`'s `delete` method
- changed serializer for error `400` in `CreateShortenedLink`'s `post` method

**Comment by Vineet-Sharma29** [comment]
@aldenstpage fixed style test error

**Comment by Vineet-Sharma29** [comment]
@aldenstpage removed unused imports from:-

Removed unused import from:-

    analytics/tests.py
    cccatalog-api/cccatalog/api/controllers/search_controller.py
    cccatalog-api/cccatalog/api/serializers/image_serializers.py
    cccatalog-api/cccatalog/api/views/image_views.py
    cccatalog-api/cccatalog/api/views/link_views.py
    cccatalog-api/cccatalog/scripts/thumbnail_load_test/locustfile.py
    cccatalog-api/cccatalog/urls.py
    ingestion_server/ingestion_server/qa.py
    ingestion_server/test/unit_tests.py
    misc/gen_popularity.py

**Comment by Vineet-Sharma29** [comment]
Build seems to work fine for my fork as shown here:-
Screenshot_2020-03-17 Build #1 - Vineet-Sharma29 cccatalog-api - Travis CI

But PR is failing build after `./load_sample_data.sh`.
**Build for PR**
Screenshot_2020-03-17 Build #1350 - creativecommons cccatalog-api - Travis CI

**Build for my fork**
Screenshot_2020-03-17 Build #1 - Vineet-Sharma29 cccatalog-api - Travis CI(1)

After observing two screenshots it can be seen that command after `load_sample_data.sh` are different.

**Comment by aldenstpage** [comment]
See https://github.com/creativecommons/cccatalog-api/issues/369 – sometimes it's necessary to restart the build.

**Comment by Vineet-Sharma29** [comment]
@aldenstpage  thanks!

File: cccatalog-api/cccatalog/api/views/image_views.py [hunk 1]
```diff
@@ -61,6 +61,8 @@ class SearchImages(APIView):
                          responses={
                              200: ImageSearchResultsSerializer(many=True),
                              400: InputErrorSerializer,
+                             500: 'relation "content_provider" does not \
```
Comment (aldenstpage) [inline cccatalog-api/cccatalog/api/views/image_views.py]: An error `500` is a generic server error, it won't necessarily be this specific one.

File: cccatalog-api/cccatalog/api/views/image_views.py [hunk 2]
```diff
@@ -143,7 +146,8 @@ class ImageDetail(GenericAPIView, RetrieveModelMixin):
                                                " particular image ID.",
                          responses={
                              200: ImageSerializer,
-                             404: 'Not Found'
+                             404: 'Not Found',
+                             500: 'relation "image" does not exist'
```
Comment (aldenstpage) [inline cccatalog-api/cccatalog/api/views/image_views.py]: see above

File: cccatalog-api/cccatalog/api/views/site_views.py [hunk 3]
```diff
@@ -136,7 +138,10 @@ class Register(APIView):
     @swagger_auto_schema(operation_id='register_api_oauth2',
                          request_body=OAuth2RegistrationSerializer,
                          responses={
-                             201: OAuth2RegistrationSuccessful
+                             201: OAuth2RegistrationSuccessful,
+                             400: OAuth2RegistrationSerializer,
+                             500: 'relation "api_oauth2registration" \
```
Comment (aldenstpage) [inline cccatalog-api/cccatalog/api/views/site_views.py]: Replace with a generic server error

File: cccatalog-api/cccatalog/api/views/site_views.py [hunk 4]
```diff
@@ -51,7 +51,9 @@ class ImageStats(APIView):
     """
     @swagger_auto_schema(operation_id='image_stats',
                          responses={
-                             200: AboutImageResponse(many=True)
+                             200: AboutImageResponse(many=True),
+                             404: 'Not Found',
+                             500: 'relation "content_provider" does not exist'
```
Comment (aldenstpage) [inline cccatalog-api/cccatalog/api/views/site_views.py]: Replace with a generic server error

File: cccatalog-api/cccatalog/api/views/link_views.py [hunk 5]
```diff
@@ -59,7 +59,9 @@ class ResolveShortenedLink(APIView):
                          responses={
                              200: None,
                              301: 'Moved Permanently',
-                             404: 'Not Found'
+                             400: ShortenedLinkSerializer,
+                             404: 'Not Found',
+                             500: 'relation "api_shortenedlink" does not exist'
```
Comment (aldenstpage) [inline cccatalog-api/cccatalog/api/views/link_views.py]: Replace with a generic server error

File: cccatalog-api/cccatalog/api/views/link_views.py [hunk 6]
```diff
@@ -22,7 +21,8 @@ class CreateShortenedLink(GenericAPIView):
     @swagger_auto_schema(operation_id="link_create",
                          responses={
                              201: _LinkCreatedResponse,
-                             400: "Bad Request"
+                             400: ShortenedLinkSerializer,
+                             500: 'relation "api_shortenedlink" does not exist'
```
Comment (aldenstpage) [inline cccatalog-api/cccatalog/api/views/link_views.py]: Replace with a generic server error

**Comment by Vineet-Sharma29** [comment]
@aldenstpage in `image_serializer` I have used `UUID field` for identifier in `ImageDetailInputErrorSerializer`. Should i change it to `CharField` as done in `ImageSerializer`?

**Comment by aldenstpage** [comment]
The PR has gotten too stale to merge without a lot of work; we'll cherry pick the changes that document error messages (and credit you properly)

**Pull request closed without merging**